### PR TITLE
Optimize allunique for large sorted arrays

### LINKED ISSUE #50352: `allunique` is not optimized for sorted arrays
```julia
julia> VERSION
v"1.10.0-DEV.1599"

julia> v=rand(1:10000000, 100000000);

julia> v2=unique(v);

julia> v3=sort(v2);

julia> @time allunique(v2)
  0.699039 seconds (20.41 k allocations: 145.416 MiB, 1.10% gc time, 6.79% compilation time)
true

julia> @time allunique(v3)
  0.669185 seconds (22 allocations: 144.048 MiB)
true

julia> allunique_fast(v) = issorted(v) ? all(i -> v[i] != v[i+1], 1:length(v)-1) : allunique(v)
allunique_fast (generic function with 1 method)

julia> @time allunique_fast(v2)
  0.703277 seconds (13.51 k allocations: 145.071 MiB, 3.20% gc time, 2.91% compilation time)
true

julia> @time allunique_fast(v3)
  0.020216 seconds
true
```
```0.02 < 0.67```

Curiously, for strings, `allunique` is slower on a sorted array

```julia
julia> using Random

julia> v=map(_ -> randstring(5), 1:100000000);

julia> v2=unique(v);

julia> v3=sort(v2);

julia> @time allunique(v2)
 13.916787 seconds (21.43 k allocations: 2.251 GiB, 4.16% gc time, 0.39% compilation time)
true

julia> @time allunique(v3)
 21.688327 seconds (1.02 k allocations: 2.250 GiB, 12.60% gc time)
true

julia> @time allunique_fast(v3)
  8.787623 seconds (11.78 k allocations: 882.406 KiB, 0.26% compilation time)
true
```

### PR DESCRIPTION
**Pull request by AshlinHarris**
Fix  #50352

### CODE REVIEW

**Comment by AshlinHarris** [comment]
Contributed code and tests were based on [this comment](https://github.com/JuliaLang/julia/issues/50352#issue-1780964726).

With this change, performance of `allunique` on large arrays is improved if the list is sorted.

Case 1: Integers
```
julia> v=rand(1:10000000, 100000000);

julia> v2=unique(v);

julia> v3=sort(v2);

julia> @time allunique(v2)
  0.686511 seconds (66.08 k allocations: 147.888 MiB, 9.99% compilation time)
true

julia> @time allunique(v3)
  0.023977 seconds
true
```

Case 2: Strings
```
julia> using Random

julia> v=map(_ -> randstring(5), 1:100000000);

julia> v2=unique(v);

julia> v3=sort(v2);

julia> @time allunique(v2)
 14.200777 seconds (55.31 k allocations: 2.253 GiB, 24.13% gc time, 0.53% compilation time)
true

julia> @time allunique(v3)
  6.636256 seconds
true

```

File: base/set.jl [hunk 1]
```diff
@@ -472,6 +472,9 @@ function allunique(C)
 end
 
 function _hashed_allunique(C)
+    if issorted(C)
+        return all(i -> C[i] != C[i+1], 1:length(C)-1)
```
Comment (ericphanson) [inline base/set.jl:476]:
```suggestion
        return all(i -> !isequal(C[i], C[i+1]), 1:length(C)-1)
```

Comment (oscardssmith) [inline base/set.jl:476]: is it valid to assume no offset indices here?

**Comment by Tortar** [comment]
Isn't there already #50372?

**Comment by AshlinHarris** [comment]
> Isn't there already #50372?

Yes, it looks like this PR is redundant. My apologies.

**Pull request closed without merging**